vars/dockerExecuteOnKubernetes.groovy

```diff
@@ -139,10 +176,10 @@ private List getContainerList(config) {
  * (Kubernetes-Plugin only!)
  * @param config Map with configurations
  */
-private List getContainerEnvs(config) {
+private List getContainerEnvs(config, imageName) {
     def containerEnv = []
-    def dockerEnvVars = config.dockerEnvVars ?: [:]
-    def dockerWorkspace = config.dockerWorkspace ?: ''
+    def dockerEnvVars = config.containerEnvVars?.get(imageName) ?: config.dockerEnvVars ?: [:]
+    def dockerWorkspace = config.containerWorkspaces?.get(imageName) != null ? config.containerWorkspaces?.get(imageName) : config.dockerWorkspace ?: ''
```

Comment:
use elvis `?:`

Reply:
not possible here, otherwise behavior would change unexpectedly